### DISCUSSION TITLE
Could someone tell me what this means

### DISCUSSION THREAD

**Discussion by superspaceshuttlee** [mrdbourke/pytorch-deep-learning#1115] [Q&A]
## **Create a linear regression model class**
```
class LinearRegressionModelV1(torch.nn.Module):
    def init(self):
        super().init()
        self.weights = torch.nn.Parameter(torch.randn(1,
                                                      requires_grad=True,
                                                      dtype=torch.float))
        self.bias = torch.nn.Parameter(torch.randn(1,
                                         dtype = torch.float))
        def forward(self, x: torch.Tensor) -> torch.Tensor:
            return self.weights * x + self.bias
```
            
I was watching the youtube tutorial and got stuck here
I don't understand what any of this code means but understood everything else before

**Comment by LuluW8071** [comment]
In simpler way we are creating a class for our model **Linear Regression** through formula

$$y = m*x + c$$

- **`__init__`** is a constructor of the class where we initialize the parameter: __weights__ ($m$) and __bias__  ($c$) 
- **`forward`** is a method defines how the model should calculate output ($y$) when given __weights__ ($m$) and __bias__  ($c$) as initialized parameter and $x$ as input data

**Reply by superspaceshuttlee** [reply]
Thanks
